internal/source/compose/utils.go

```diff
@@ -31,6 +42,157 @@ const (
 	envFile               string = "env_file"
 )
 
+// IsV3 returns if the docker-compose yaml is version 3
+func IsV3(path string) (bool, error) {
+	dcfile := sourcetypes.DockerCompose{}
+	if err := common.ReadYaml(path, &dcfile); err != nil {
+		log.Errorf("Unable to read docker compose yaml at path %s Error: %q", path, err)
+	}
+	log.Debugf("Docker Compose version: %s", dcfile.Version)
+	switch dcfile.Version {
+	case "", "1", "1.0", "2", "2.0", "2.1":
+		return false, nil
+	case "3", "3.0", "3.1", "3.2", "3.3", "3.4", "3.5", "3.6", "3.7", "3.8":
+		return true, nil
+	default:
+		err := fmt.Errorf("The compose file at path %s uses Docker Compose version %s which is not supported. Please use version 1, 2 or 3", path, dcfile.Version)
+		log.Errorf("Error: %q", err)
+		return false, err
+	}
+}
+
+// ParseV2 parses version 2 compose files
```

Comment:
Would it not be better to move parseV2 and parseV3 functions to the respective version files instead of putting it in utils.go?